scipy/spatial/distance.py

```diff
@@ -121,6 +121,40 @@
 from . import _hausdorff
 from ..linalg import norm
 
+def _args_to_kwargs_xdist(args, kwargs, metric, func_name):
+    if func_name == "pdist":
+        old_arg_names = ["p", "w", "V", "VI"]
+    else:
+        old_arg_names = ["p", "V", "VI", "w"]
+    num_args = len(args)
+    if num_args:
+        if(callable(metric) and
+           metric not in [braycurtis, canberra, chebyshev, cityblock,
+                          correlation, cosine, dice, euclidean,
+                          hamming, jaccard, kulsinski, mahalanobis,
+                          matching, minkowski, rogerstanimoto,
+                          russellrao, seuclidean, sokalmichener,
+                          sokalsneath, sqeuclidean, yule, wminkowski]):
+            raise TypeError('When using a custom metric arguments must be passed'
+                            'as keyword (i.e., ARGNAME=ARGVALUE)')
+
+        warnings.warn('%d metric parameters have been passed as positional.'
+                      'This will raise an error in a future version.'
+                      'Please pass arguments as keywords'
+                      '(i.e., ARGNAME=ARGVALUE)' % num_args, DeprecationWarning)
+        if num_args > 4:
+            raise ValueError('Deprecated %s signature accepts only 4'
+                'positional arguments (p, w, V, VI), %d given.'
+                % (func_name, num_args))
+
+        for i, arg in enumerate(args):
+            if old_arg_names[i] in kwargs:
+                raise TypeError('%s() got multiple values for argument '
+                    '%s' % (func_name, old_arg_names[i]))
+            kwargs[old_arg_names[i]] = arg
+    return kwargs
+
```

Comment:
This function is meant to be temporary. As soon as the *args are removed this can safely removed too